android/java/org/chromium/chrome/browser/crypto_wallet/fragments/ApproveTxBottomSheetDialogFragment.java

```diff
@@ -0,0 +1,208 @@
+
+        dialog.setContentView(view);
+        ViewParent parent = view.getParent();
+        ((View) parent).getLayoutParams().height = ViewGroup.LayoutParams.WRAP_CONTENT;
+        TextView networkName = view.findViewById(R.id.network_name);
+        networkName.setText(mNetworkName);
+        ImageView icon = (ImageView) view.findViewById(R.id.account_picture);
+        icon.setImageResource(mAccountPic);
+        TextView fromTo = view.findViewById(R.id.from_to);
+        fromTo.setText(
+                mAccountName + " -> " + Utils.stripAccountAddress(mTxInfo.txData.baseData.to));
+        TextView txType = view.findViewById(R.id.tx_type);
+        txType.setText(mTxType);
+        TextView amountAsset = view.findViewById(R.id.amount_asset);
+        amountAsset.setText(String.format("%.4f", Utils.fromHexWei(mTxInfo.txData.baseData.value))
+                + " " + mAsset);
+        AssetRatioController assetRatioController = getAssetRatioController();
+        if (assetRatioController != null) {
+            String[] assets = {mAsset.toLowerCase()};
+            String[] toCurr = {"usd"};
+            assetRatioController.getPrice(
+                    assets, toCurr, AssetPriceTimeframe.LIVE, (success, values) -> {
+                        if (!success || values.length == 0) {
+                            return;
+                        }
+                        double value = Utils.fromHexWei(mTxInfo.txData.baseData.value);
+                        double price = Double.valueOf(values[0].price);
+                        mTotalPrice = value * price;
+                        TextView amountFiat = view.findViewById(R.id.amount_fiat);
+                        amountFiat.setText("$" + String.format("%.2f", mTotalPrice));
+                        ViewPager viewPager = view.findViewById(R.id.navigation_view_pager);
+                        ApproveTxFragmentPageAdapter adapter =
+                                new ApproveTxFragmentPageAdapter(getChildFragmentManager(), mTxInfo,
+                                        mAsset, mTotalPrice, getActivity());
+                        viewPager.setAdapter(adapter);
+                        viewPager.setOffscreenPageLimit(adapter.getCount() - 1);
+                        TabLayout tabLayout = view.findViewById(R.id.tabs);
+                        tabLayout.setupWithViewPager(viewPager);
+                    });
+        }
+
+        Button reject = view.findViewById(R.id.reject);
+        reject.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                rejectTransaction(true);
+            }
+        });
+
+        Button approve = view.findViewById(R.id.approve);
+        approve.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                EthTxController ethTxController = getEthTxController();
```

Comment:
For reject button processing is moved to separate method, but for approve button processing is directly inside handler. 

Maybe it worth to make similar `private void approveTransaction()` ?

Reply:
I can move it yes. The reject was moved to a separate function as we have to call it from 2 places.